google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugProcessStateCollector.java

```diff
@@ -0,0 +1,96 @@
+/*
```

Comment:
How is testing made easier?

And I disagree that functionality is independent. I would expect that a CloudDebugGlobalPoller polls everything - snapshots and states in this case. What we could want to do is to rename CloudDebugGlobalPoller to something else (like CloudBreakpointPoller/Collector), but you might want to leave a TODO for another PR.